### DISCUSSION TITLE
useFieldArray shouldUnregister false + Swap bug?

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#3076] [Q&A] [answered]
Hello, there seems to be an issue w/ should unregister false and swap.

v6.9.0
Click append twice, then click swap. One of the array items of the field becomes undefined and leads to a crash.
https://codesandbox.io/s/react-hook-form-690-usefieldarray-swap-ws3r2?file=/src/index.js

v6.8.6 but w/ insert instead of append (my old logic)
Click append (insert) twice, then click swap. No crash and actually swaps.
https://codesandbox.io/s/react-hook-form-686-usefieldarray-swap-forked-gmfw6?file=/src/index.js

**Comment by bluebill1049** [comment]
thanks i will take a look at this issue.

**Comment by bluebill1049** [comment, marked as answer]
tested the previous version is broken as well, let me find out the root cause.

**Reply by bluebill1049** [reply]
there is no input in your useFieldArray? we need input to get registered.

**Reply by fadi-george** [reply]
I could potentially add hidden inputs but It worked fine before w/o registered inputs with the insert logic above.

**Reply by bluebill1049** [reply]
i will take a look again, i guess this is duplicated? https://github.com/react-hook-form/react-hook-form/discussions/3075

**Reply by bluebill1049** [reply]
patching this: https://github.com/react-hook-form/react-hook-form/issues/3080 but i wouldn't recommend usage without input. 👍

**Reply by fadi-george** [reply]
I'm using a modal for the user to generate the data and pass it to the field array. But thank you very much for looking into it!

**Comment by fadi-george** [comment]
#3075 is a separate issue.
